src/main/java/org/javacord/bot/commands/WikiCommand.java

```diff
@@ -2,23 +2,123 @@
+                    populatePages(channel.getApi(), embed, titleOnly(searchString));
+                } else if (args[0].matches("(full|f|content|c)")) {
+                    populatePages(channel.getApi(), embed,
+                            titleOnly(searchString).or(keywordsOnly(searchString)).or(contentOnly(searchString)));
+                } else {
+                    searchString = String.join(" ", Arrays.copyOfRange(args, 0, args.length)).toLowerCase();
+                    populatePages(channel.getApi(), embed, titleOnly(searchString).or(keywordsOnly(searchString)));
+                }
+                channel.sendMessage(embed).join();
+            }
+        } catch (Throwable t) {
+            channel.sendMessage("Something went wrong: ```" + ExceptionLogger.unwrapThrowable(t).getMessage() + "```").join();
+            // Throw the caught exception again. The sdcf4j will log it.
+            throw t;
+        }
+    }
+
+    private Predicate<WikiPage> titleOnly(String searchString) {
+        return p -> p.getTitle().toLowerCase().contains(searchString);
+    }
+
+    private Predicate<WikiPage> keywordsOnly(String searchString) {
+        return p -> Arrays.stream(p.getKeywords())
+                .map(String::toLowerCase)
+                .anyMatch(k -> k.contains(searchString));
+    }
+
+    private Predicate<WikiPage> contentOnly(String searchString) {
+        return p -> p.getContent().toLowerCase().contains(searchString);
+    }
+
+
+    private void populatePages(DiscordApi api, EmbedBuilder embed, Predicate<WikiPage> searchCriteria) {
+        List<WikiPage> pages = new WikiParser(api)
+                .getPages().join().stream()
+                .filter(searchCriteria)
+                .sorted()
+                .collect(Collectors.toList());
+        if (pages.isEmpty()) {
+            embed.setTitle("Javacord Wiki");
+            embed.setUrl(WikiParser.BASE_URL + "/wiki/");
+            embed.setDescription("No pages found. Maybe try another search.");
+            embed.addField("Standard Search", "Use `!wiki <search>` to search page titles and keywords.");
+            embed.addField("Title Search", "Use `!wiki [page|p|title|t] <search>` to exclusively search page titles.");
+            embed.addField("Full Search", "Use `!wiki [full|f|content|c] <search>` to perform a full search.");
+        } else if (pages.size() == 1) {
+            WikiPage page = pages.get(0);
+            embed.setUrl(WikiParser.BASE_URL + page.getUrl());
+            embed.setTitle("Wiki: " + page.getTitle());
+            String cleanedDescription = page.getContent().replaceAll("<[^>]+>", "").trim();
+            int length = 0;
+            int sentences = 0;
+            while (length < 600 && sentences < 3) {
+                int tmpLength = cleanedDescription.indexOf(". ", length);
+                length = (tmpLength > length) ? tmpLength : cleanedDescription.indexOf(".\n");
+
+                sentences++;
+            }
```

Comment:
If content is cropped, some continuation marker would be nice, so see that there is more if you open the actual page.